src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java

```diff
@@ -699,6 +697,16 @@ private Position decodeLocation(DeviceSession deviceSession, ByteBuf buf) {
                         }
                     }
                     break;
+                case 0xF4:
+                    while (buf.readerIndex() < endIndex) {
+                        WifiAccessPoint wifiAccessPoint = new WifiAccessPoint();
+                        wifiAccessPoint.setMacAddress(String.format("%02x:%02x:%02x:%02x:%02x:%02x",
+                                buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte(),
+                                buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte()));
+                        wifiAccessPoint.setSignalStrength((int) buf.readByte());
+                        network.addWifiAccessPoint(wifiAccessPoint);
```

Comment:
Hopefully the last thing, but there's a more concise way of doing this. Something like this:
```java
                            String mac = ByteBufUtil.hexDump(buf.readSlice(6)).replaceAll("(..)", "$1:");
                            network.addWifiAccessPoint(WifiAccessPoint.from(
                                    mac.substring(0, mac.length() - 1), buf.readByte()));
```
We do it in many other places.